lib/mocha.js

```diff
@@ -1212,29 +1214,45 @@ Mocha.prototype.runGlobalTeardown = async function runGlobalTeardown(
   const {globalTeardown} = this.options;
   if (globalTeardown && globalTeardown.length) {
     debug('run(): global teardown starting');
-    await this._runGlobalFixtures(globalTeardown, context);
+    await this._runGlobalFixtures(globalTeardown, context, "Global Teardown");
   }
   debug('run(): global teardown complete');
   return context;
 };
 
 /**
- * Run global fixtures sequentially with context `context`
+ * Run global fixtures sequentially with context `context`.
  * @private
  * @param {MochaGlobalFixture[]} [fixtureFns] - Fixtures to run
  * @param {object} [context] - context object
+ * @params {string} [phase] - phase of the fixture
  * @returns {Promise<object>} context object
  */
 Mocha.prototype._runGlobalFixtures = async function _runGlobalFixtures(
   fixtureFns = [],
-  context = {}
+  context = {},
+  phase
 ) {
-  for await (const fixtureFn of fixtureFns) {
-    await fixtureFn.call(context);
+  for (const fixtureFn of fixtureFns) {
+    try {
+      await fixtureFn.call(context);
+    } catch (err) {
+
+      context.stats.failures++;
+      context.failures++;
+        
+      context.emit('fail', {
+        title: phase,
+        err: err
+      });
+      console.error(err);
+      process.exit(1);
```

Comment:
[Bug] We can't manually `process.exit(1)` ourselves. Someone might be calling the Mocha API manually and expect to get results returned back to them. If you want to mark all tests as failed, you'll have to find another way.